docs/cpp_docs.rst

```diff
@@ -26,94 +44,556 @@ The beginning of both header and source code files should include
+    */
+   template <class Func>
+   ImageF transformImage(Func const & transform) const;
+
+.. _cpp-doxygen-related:
+
+Helper Functions
+----------------
+
+Some operations on a class, particularly arithmetic operations, must be implemented as standalone functions even though they are *conceptually* part of a class. These functions should have the ``@relatesalso`` tag, followed by the name of the appropriate class. They will appear on the class's documentation page under the heading "Related Functions". Use this tag sparingly.
+
+For internal consistency, always use ``@relatesalso`` and not the synonymous ``@relatedalso``.
+
+.. _cpp-doxygen-initializer:
+
+Initializer Declaration
+-----------------------
+
+By default, Doxygen shows the values of constants unless they are very long. The ``@showinitializer`` and ``@hideinitializer`` tags override this behavior.
+
+.. _cpp-doxygen-see-also:
+
+See Also
+--------
+
+'See Also' is an optional section used to refer to related code.
+This section can be very useful, but should be used judiciously.
+The goal is to direct users to other functions they may not be aware of, or have easy means of discovering (by looking at the class or package documentation, for example).
+Functions whose documentation further explains parameters used by this function are good candidates.
+
+List each class, function, or method using a ``@see`` tag, followed by a ``@ref`` tag:
+
+.. code-block:: cpp
+
+   /**
+    * Compute an element-wise cosine.
+    *
+    * @see @ref sin
+    * @see @ref tan
+    */
+   vector<double> cos(vector<double> const & angles);
+
+For internal consistency, always use ``@see`` and not the synonymous ``@sa``.
+
+.. _cpp-doxygen-notes:
+
+Notes
+-----
+
+*Notes* is an optional section that provides additional information about the code, possibly including a discussion of the algorithm or known limitations of the code. The notes should be prefixed by a ``@note`` or ``@warning`` command.
+
+For internal consistency, always use ``@note`` and not the synonymous ``@remark`` or ``@remarks``.
+
+This section may include mathematical equations, written in `LaTeX <http://www.latex-project.org/>`_ format:
+
+.. code-block:: cpp
+
+   /**
+    * The FFT is a fast implementation of the discrete Fourier transform:
+    * @f[ X(e^{j\omega } ) = x(n)e^{ - j\omega n} @f]
+    */
+
+LaTeX environments can also be used:
+
+.. code-block:: cpp
+
+   /**
+    * The discrete-time Fourier time-convolution property states that
+    * @f{eqnarray*}
+    * x(n) * y(n) \Leftrightarrow X(e^{j\omega } )Y(e^{j\omega } )\\
+    * another equation here
+    * @f}
+    */
+
+Math can also be used inline:
+
+.. code-block:: cpp
+
+   /**
+    * The value of @f$\omega@f$ is larger than 5.
+    */
+
+Note that LaTeX is not particularly easy to read, so use equations sparingly. Doxygen recovers poorly from typos in formulas; you may need to manually delete ``docs/html/formula.repository`` if it contains a bad formula.
```

Comment:
I slightly disagree with this.  Latex markup is indeed hard to read, but it can make the rendered documentation vastly easier to read, especially for the hardest-to-explain code objects.  Perhaps we could reword this to discourage gratuitous use of Latex math (e.g. to get script Greek letters) while encouraging it for highly mathematical code objects?